wrapanapi/lenovo.py

```diff
@@ -18,25 +22,62 @@ class LenovoSystem(WrapanapiAPIBase):
     _api = None
 
     _stats_available = {
-        'num_server': lambda self: len(self.list_server()),
+        'num_server': lambda self: len(self.list_servers()),
+        'num_server_with_host': lambda self: len(self.list_servers_with_host()),
     }
+    POWERED_ON = 8
+    POWERED_OFF = 5
+    STANDBY = 18
+    HEALTH_VALID = ("normal", "non-critical")
+    HEALTH_WARNING = ("warning")
+    HEALTH_CRITICAL = ("critical", "minor-failure", "major-failure", "non-recoverable", "fatal")
 
-    def __init__(self, hostname, username, password, **kwargs):
-        super(LenovoSystem, self).__init__(kwargs)
+    def __init__(self, hostname, username, password, protocol="https", port=443, **kwargs):
         self.hostname = hostname
+        self.port = port
         self.username = username
         self.password = password
+        self.protocol = protocol
 
     def __del__(self):
         """Disconnect from the API when the object is deleted"""
         # This isn't the best place for this, but this class doesn't know when it is no longer in
         # use, and we need to do some sort of disconnect based on the pyVmomi documentation.
-        raise NotImplementedError
+
+    def __service_instance(self, path):
+        """An instance of the service"""
+        try:
+            uri = urlunparse((self.protocol, self.hostname, path, "", "", ""))
+            response = requests.get(uri, auth=(self.username, self.password), verify=False)
+            return json.loads(response.content)
+        except Timeout:
+            return None
 
     @property
     def version(self):
         """The product version"""
-        raise NotImplementedError
+        response = self.__service_instance("/aicc")
+        return response['appliance']['version']
+
+    def list_servers(self):
+        response = self.__service_instance("/cabinet?status=includestandalone")
+        cabinets = response['cabinetList'][0]
+        nodes_list = cabinets['nodeList']
+
+        inventory_list = map(lambda x: x['itemInventory'], nodes_list)
```

Comment:
consider using a generator here instead of lambda

`inventory_list = [node['itemInventory'] for node in nodes_list]`